components/FileInput.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useState } from 'react';
+import { useDropzone, DropzoneProps } from 'react-dropzone';
+import { Controller, Control } from 'react-hook-form';
+import Image from 'next/image';
+import Box from '@mui/material/Box';
+import Typography from '@mui/material/Typography';
+
+interface FileInputProps extends DropzoneProps {
+  name: string;
+  instructions: string;
+  selectedFileMessage: string;
+  control: Control;
+  rules?: object;
+}
+
+const FileInput = ({
+  name,
+  instructions,
+  selectedFileMessage,
+  control,
+  accept,
+  maxSize,
+  minSize,
+  multiple,
+  rules,
+}: FileInputProps) => {
+  const [
+    filePreview,
+    setFilePreview,
+  ] = useState<string | null>(null);
+
+  return (
+    <Controller
+      name={name}
+      control={control}
+      rules={rules}
+      render={({ field: { onChange }, fieldState: { error } }) => {
+        const handleDrop = (acceptedFiles: File[]) => {
+          onChange(acceptedFiles.length > 0 ? acceptedFiles[0] : null);
+          if (acceptedFiles[0] && acceptedFiles[0].type.startsWith('image/')) {
+            setFilePreview(URL.createObjectURL(acceptedFiles[0]));
+          } else {
+            setFilePreview(null);
+          }
+        };
+
+        const {
+          getRootProps: getCustomRootProps,
+          getInputProps: getCustomInputProps,
+          isDragActive,
+        } = useDropzone({
```

Comment:
If you are using `useDropzone` approach, then it's better to create a separate component where your `handleDrop` and `useDropzone` are defined before the `return` statement. You can pass necessary props, such as `onChange` and `error` to this component.